src/Data/Monoid/Inf.hs

```diff
@@ -75,6 +76,51 @@ instance Ord a => Monoid (Inf Neg a) where
   mempty = Infinity
   mappend = (<>)
 
+instance Applicative (Inf p) where
+    pure = Finite
+    Infinity <*> _ = Infinity
+    _ <*> Infinity = Infinity
+    Finite f <*> Finite x = Finite $ f x
+
+instance Monad (Inf p) where
+    Infinity >>= _ = Infinity
+    Finite x >>= f = f x
+
+instance Bounded a => Bounded (NegInf a) where
+    minBound = Infinity
+    maxBound = Finite maxBound
+
+instance Bounded a => Bounded (PosInf a) where
+    minBound = Finite minBound
+    maxBound = Infinity
+
+instance Num a => Num (Inf p a) where
```

Comment:
Sure, those instances make sense, it's just `Num` and friends that I don't like.  Maybe the simplest thing is just to close this PR and open a new one with only the other instances added. Would you be willing to do that?